### PR TITLE
Do not look for XML files in child directories

### LINKED ISSUE #899: Force field loaded from nested child directory
**Describe the bug**

It seems like #816 introduces a bug whereby a `ForceField` can inadvertently be loaded from a nested child of the current working directory. This can presumably lead to subtle issues whereby the user may think they are loading a force field in their current working directory, but if they forgot to create / copy a file there, load a force field from a child directory with the same name but not be the intended force field.

This issue is related to #898 

**To Reproduce**

```
import os

from openff.toolkit.typing.engines.smirnoff import ForceField

nested_directory = os.path.join("a", "b", "c")
os.makedirs(nested_directory, exist_ok=True)

# Create a FF in a nested directory
ForceField("openff-1.3.0.offxml").to_file(
    os.path.join(nested_directory, "force-field.offxml")
)

# Check that the file does not exist in the current working directory.
assert not os.path.isfile("force-field.offxml")

print(ForceField("force-field.offxml"))
```

**Output**

```
<openff.toolkit.typing.engines.smirnoff.forcefield.ForceField object at 0x7fa44f45acd0>
```

**Computing environment (please complete the following information):**
 - Operating system: OSX
 - Output of running `conda list` 

```
openff-toolkit            0.9.1              pyhd8ed1ab_1    conda-forge
openff-toolkit-base       0.9.1              pyhd8ed1ab_1    conda-forge
python                    3.9.2           h2502468_0_cpython    conda-forge
```

**Additional context**

### PR DESCRIPTION
**Pull request by mattwthompson**
This PR involves
1. Taking the concise MWE from https://github.com/openforcefield/openff-toolkit/issues/899#issue-848178967 and making it a unit test
2. Reverting the extremely aggressive `.glob("**/*.offxml")`, which looks _recursively into all subdirectories_ for anything with an `.offxml` extenstion. https://github.com/openforcefield/openff-toolkit/issues/898 also explains why the current implementation (mine) was not the right approach.

This should directly fix  #899. #898 should be fixed by extension, although it would be tricky to test; I'd expect a file in a deleted subdirectory to not be looked at if files in subdirectories are not looked at in the first place.

Note that the check mark from ed00e65 is misleading, since tests didn't run on that commit. The point of committing in that order was to show the broken test and then how it's fixed, but that didn't work.

- [x] Tag issue being addressed
- [x] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/master/openff/toolkit/tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/master/docs), if applicable
- [x] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [ ] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/master/docs/releasehistory.rst)

### CODE REVIEW

File: openff/toolkit/tests/test_forcefield.py [hunk 1]
```diff
@@ -931,6 +931,25 @@ def test_force_field_case(self):
 
         assert hash(default_case) == hash(lower_case)
 
+    def test_do_not_load_in_child_dir(self):
```
Comment (j-wags) [inline openff/toolkit/tests/test_forcefield.py:934]: Could we do this test in a temp dir? Right now it leaves the directories around.
```suggestion
    def test_do_not_load_in_child_dir(self, tmp_path):
```

**Review by j-wags — APPROVED** [review]
Looks great. Thanks for putting this together, @mattwthompson! Small suggestion, also please update the releasehistory before you merge :-)

File: openff/toolkit/tests/test_forcefield.py (hunk 1 above)
Comment (mattwthompson) [reply openff/toolkit/tests/test_forcefield.py:934]: Good catch - I wonder how many other times our tests leave temporary directories/files/souvenirs behind?

Comment (mattwthompson) [reply openff/toolkit/tests/test_forcefield.py]: Hmm, seems like that didn't quite do it:

```
$ # git clone https://github.com/openforcefield/openff-toolkit
$ # cd openff-toolkit
$ # conda activate openff-dev
$ # python -m pytest -v -r fE --tb=short --ignore=openff/toolkit/tests/test_examples.py  --ignore=openff/toolkit/tests/test_links.py --runslow openff/toolkit/tests
$ git status
git status                                                                                                                                         master  ✱
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	a/
	openff/toolkit/data/molecules/AlkEthOH_tripos/
	openff/toolkit/data/molecules/FreeSolv/
```

Comment (j-wags) [reply openff/toolkit/tests/test_forcefield.py]: Hm, I don't remember the exact convention with this, but we may need to have a context manager `temporary_cd` into that directory as well.

Comment (lilyminium) [reply openff/toolkit/tests/test_forcefield.py]: Yeah, just creating the tmp_path doesn't go to it. I use this construction a lot:

```python
def testy_test(tmpdir):
	with tmpdir.as_cwd():
		print(os.cwd())
```

**Pull request merged**